_posts/2021-01-11-zappa-serverless-docker.md

```diff
@@ -0,0 +1,224 @@
+
+Last month, [AWS announced new functionality](https://aws.amazon.com/blogs/aws/new-for-aws-lambda-container-image-support/) that lets you deploy Lambda functions as **container images** up to 10GB in size. What does that mean? Instead of pre-compiling all your packages on Amazon Linux, putting them & your application code in a zip file and hoping you are under the limit, you can now just create a Docker image and deploy your function with that. 
+
+Today, I'm happy to announce that you **can now use Zappa to manage your serverless deployments with Docker ✨🍰✨** .
+
+# hello Docker + Zappa!
+
+With the [latest release]() of Zappa, you can now deploy and/or update your Lambda functions with a Docker image. Here's a quick preview of me doing just that:
+
+<p align="center">
+    <img src="{{ site.baseurl }}{% link images/zappa-serverless-docker/zappa_docker.gif %}">
+</p>
+
+***What's going on here?***
+
+1. I call `zappa deploy` and supply a Docker image URI to the `-d / --docker-image-uri` parameter. I'm using a prebuilt Docker image that lives in an AWS [Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) repository. In a future release of Zappa, we will add in functionality to automatically build the Docker image & push it to ECR so you don't need to worry about these steps.
+2. Zappa creates the new Lambda function with that Docker image, and attaches a new API Gateway endpoint to it.
+3. I make a few web requests to new API endpoint, they invoke the Lambda function and get executed in the Flask app running in my Docker image.
+
+For the remainder of the post, we'll dive into the details that make this work.
+
+# Making it happen
+
+## The App
+
+The web app we're deploying is a simple [Flask](https://flask.palletsprojects.com/en/1.1.x/) app with two endpoints, `/` and `/time`.
+
+```python
+# zdf/app.py
+import time
+from flask import Flask, jsonify
+
+app = Flask(__name__)
+
+@app.route("/")
+def serve():
+    return jsonify(success=True)
+
+@app.route("/time")
+def get_current_time():
+    return {"time": round(time.time())}
+```
+
+We'll also have a batch process we run at regular intervals:
+
+```python
+# zdf/process.py
+from datetime import datetime
+
+def run_process():
+    now_str = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
+    print(f"The current time is {now_str}")
+```
+
+## zappa_settings.json
+
+Our `zappa_settings.json` is pretty minimal, and doesn't require any changes from your typical one. The primary difference is you don't need to specify a Python runtime since that will be set in the Docker image.
```

Comment:
```suggestion
Our `zappa_settings.json` is pretty minimal, and **no different from the zappa_settings you're used to**. The only thing worth mentioning is you don't need to specify a Python runtime since that will be set in the Docker image.
```